src/execution_plan/optimizations/utilize_indices.c

```diff
@@ -508,6 +514,56 @@ void reduce_cond_op(ExecutionPlan *plan, OpCondTraverse *cond) {
 	array_free(filters);
 }
 
+// try to replace given ExpandInto Traverse operation and a set of Filter
+// operations with a single Index Scan operation
+void reduce_expand_into_op
```

Comment:
double check this

Reply:
Good call! turning an expand-into operation into an edge index scan hurt the performance dramatically. Reverted!